### PR TITLE
feat(portex): add PortexRecordBase for Portex types based on record

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2g5p6nj 增加 RecordBase 基类，包含对fields进行统一操作的接口](https://app.clickup.com/t/2g5p6nj)

File: graviti/portex/base.py [hunk 1]
```diff
@@ -235,6 +248,77 @@ def copy(self: _T) -> _T:
...
+
+    @property
+    # def _fields(self) -> "ConnectedFields":
+    def _fields(self) -> NameOrderedDict[PortexType]:
+        return self._fields_factory(  # type: ignore[no-any-return]
+            {name: getattr(self, name) for name in self.params}
+        )
+
+    def __len__(self) -> int:
+        return self._fields.__len__()
```
Comment (linjiX) [inline graviti/portex/base.py:265]: use `UserMutableMapping`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/portex/base.py [hunk 2]
```diff
@@ -235,6 +247,63 @@ def copy(self: _T) -> _T:
         return deepcopy(self)
 
 
+class PortexRecordBase(
+    PortexType, UserMutableMapping[str, PortexType]
+):  # pylint: disable=abstract-method
```
Comment (linjiX) [inline graviti/portex/base.py:252]: why disable abstract-method?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/portex/base.py (hunk 2 above)
Comment (Lee-000) [reply graviti/portex/base.py:252]: Because `_from_pyarrow` is not implemented.

Comment (Lee-000) [reply graviti/portex/base.py:252]: Because `_from_pyarrow` is not implemented.

**Review by linjiX — APPROVED** [review]

**Pull request merged**